src/main.rs

```diff
@@ -119,7 +145,26 @@ fn do_main() -> Result<(), Error> {
     }
 }
 
+fn sudo_if_not_root() -> bool {
+    let euid = nix::unistd::Uid::effective();
+    if euid.is_root() {
+        return true;
+    } else {
+        println!("rbspy only works as root on Mac. Try rerunning with `sudo --preserve-env !!`.");
+        println!(
+            "If you run `sudo rbspy record ruby your-program.rb`, rbspy will drop privileges when running `ruby your-program.rb`. If you want the Ruby program to run as root, use `rbspy --no-drop-root`."
+        );
+        return false;
+    }
+}
+
 fn main() {
+    if cfg!(target_os = "macos") {
```

Comment:
alternatively, have a `#[cfg(target_os = "macos")]` and `#[cfg(not(target_os = "macos"))]` version of `sudo_if_not_root`, where the second just returns true. maybe change the name in that case though.